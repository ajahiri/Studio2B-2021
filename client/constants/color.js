export const black = '#141414';
export const gray = '#505050';
export const lightGray = '#e1e1e1';
export const darkGray = '#191919';
export const white = '#fafafa';

export const accent = '#3d3abf';
export const accentFocused = '#312e99';
export const accentDisabled = '#6461cc';

export const error = 'red';
