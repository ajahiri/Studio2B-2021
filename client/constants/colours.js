export default {
  primary: '#1e4ae7',
  primaryLight: '#a7b4e1',

  error: '#c20d0d',
  success: '#52c20d',

  white: '#e5e5e5',
  darkGrey: '#929292',
  grey: '#acacac',
  black: '#070707',
};
