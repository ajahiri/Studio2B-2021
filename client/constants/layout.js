export const spacing = {
  huge: 64,
  xxl: 32,
  xl: 24,
  lg: 16,
  md: 8,
  sm: 4,
  xs: 2,
  zero: 0,
};

export const radius = {
  lg: 16,
  md: 12,
  sm: 8,
  zero: 0,
};

export const border = {
  thick: 2,
  thin: 1,
  zero: 0,
};

export const defaultScreenMargins = {
  horizontal: spacing.xl,
  vertical: spacing.huge,
};

export const buttonSizes = {
  large: 60,
  small: 40,
};
