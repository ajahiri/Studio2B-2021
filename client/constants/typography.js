export default {
  sizes: {
    caption: 15,
    normal: 18,
    large: 20,
    subtitle: 30,
    title: 60,
  },
  weights: {
    thin: '300',
    normal: '500',
    bold: '600',
    heavy: '700',
  },
};
