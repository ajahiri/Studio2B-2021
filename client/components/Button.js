import React from 'react';
import PropTypes from 'prop-types';
import {
  ActivityIndicator,
  StyleSheet,
  Text,
  TouchableHighlight,
} from 'react-native';

import { color, font, layout } from '../constants';

export const BUTTON_CONTAINER_ID = 'button-container';
export const BUTTON_TEXT_ID = 'button-text';
export const BUTTON_ACTIVITY_INDICATOR_ID = 'button-activity-indicator';

const commonContainerStyle = {
  alignItems: 'center',
  justifyContent: 'center',
  borderWidth: 2,
  borderBottomWidth: 5,
};

function makeButtonStyles(sizeStyles, colorStyles) {
  return StyleSheet.create({
    defaultContainer: {
      ...commonContainerStyle,
      height: sizeStyles.container.height,
      borderRadius: sizeStyles.container.borderRadius,
      backgroundColor: colorStyles.container.default.backgroundColor ?? 'blue',
      borderColor: colorStyles.container.default.borderColor,
    },
    disabledContainer: {
      ...commonContainerStyle,
      height: sizeStyles.container.height,
      borderRadius: sizeStyles.container.borderRadius,
      backgroundColor: colorStyles.container.disabled.backgroundColor ?? 'blue',
      borderColor: colorStyles.container.disabled.borderColor,
    },
    defaultTitle: {
      ...sizeStyles.title,
      color: colorStyles.title.default.color,
    },
    disabledTitle: {
      ...sizeStyles.title,
      color: colorStyles.title.disabled.color,
    },
  });
}

export default function Button({
  title,
  type,
  size,
  disabled,
  isLoading,
  onPress,
  ...props
}) {
  const sizeStyles = (() => {
    switch (size) {
      case 'small':
        return {
          container: {
            height: layout.buttonSizes.small,
            borderRadius: layout.radius.md,
          },
          title: font.largeBold,
        };
      case 'large': /* FALLTHROUGH */
      default:
        return {
          container: {
            height: layout.buttonSizes.large,
            borderRadius: layout.radius.lg,
          },
          title: font.h3,
        };
    }
  })();

  const colorStyles = (() => {
    switch (type) {
      case 'primary':
        return {
          container: {
            default: {
              backgroundColor: color.yellow500,
              borderColor: color.defaultBorderColor,
            },
            focused: {
              backgroundColor: color.accentFocused,
              borderColor: color.defaultBorderColor,
            },
            disabled: {
              backgroundColor: color.accentDisabled,
              borderColor: color.disabledBorderColor,
            },
          },
          title: {
            default: {
              color: color.defaultDarkTextColor,
            },
            disabled: {
              color: color.disabledDarkTextColor,
            },
          },
        };
      case 'danger':
        return {
          container: {
            default: {
              backgroundColor: color.danger,
              borderColor: color.defaultBorderColor,
            },
            focused: {
              backgroundColor: color.dangerFocused,
              borderColor: color.defaultBorderColor,
            },
            disabled: {
              backgroundColor: color.dangerDisabled,
              borderColor: color.disabledBorderColor,
            },
          },
          title: {
            default: {
              color: color.defaultLightTextColor,
            },
            disabled: {
              color: color.disabledLightTextColor,
            },
          },
        };
      case 'secondary': /* FALLTHROUGH */
      default:
        return {
          container: {
            default: {
              backgroundColor: color.white,
              borderColor: color.defaultBorderColor,
            },
            focused: {
              backgroundColor: color.gray200,
              borderColor: color.defaultBorderColor,
            },
            disabled: {
              backgroundColor: color.white,
              borderColor: color.disabledBorderColor,
            },
          },
          title: {
            default: {
              color: color.defaultDarkTextColor,
            },
            disabled: {
              color: color.disabledDarkTextColor,
            },
          },
        };
    }
  })();

  const buttonStyles = makeButtonStyles(sizeStyles, colorStyles);

  return (
    <TouchableHighlight
      testID={BUTTON_CONTAINER_ID}
      disabled={disabled || isLoading}
      onPress={onPress}
      underlayColor={colorStyles.container.focused.backgroundColor}
      style={[
        disabled
          ? buttonStyles.disabledContainer
          : buttonStyles.defaultContainer,
        props.style,
      ]}>
      {isLoading ? (
        <ActivityIndicator
          testID={BUTTON_ACTIVITY_INDICATOR_ID}
          size="small"
          color={buttonStyles.defaultTitle.color}
        />
      ) : (
        <Text
          testID={BUTTON_TEXT_ID}
          style={[
            disabled ? buttonStyles.disabledTitle : buttonStyles.defaultTitle,
          ]}>
          {title}
        </Text>
      )}
    </TouchableHighlight>
  );
}

Button.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['primary', 'secondary', 'danger']),
  size: PropTypes.oneOf(['large', 'small']),
  disabled: PropTypes.bool,
  isLoading: PropTypes.bool,
  onPress: PropTypes.func,
};

Button.defaultProps = {
  title: '',
  type: 'secondary',
  size: 'large',
  disabled: false,
  isLoading: false,
  onPress: () => {},
};
