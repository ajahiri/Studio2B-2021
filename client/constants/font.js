import * as color from './color';

export const size = {
  h1: 60,
  h2: 30,
  h3: 25,
  h4: 20,
  lg: 18,
  md: 16,
  sm: 14,
  xs: 12,
};

export const fontFamily = {
  light: 'Inter_300Light',
  regular: 'Inter_400Regular',
  medium: 'Inter_500Medium',
  bold: 'Inter_700Bold',
};

const defaultTextStyles = {
  color: color.black,
};

export const h1 = {
  ...defaultTextStyles,
  fontSize: size.h1,
  fontFamily: fontFamily.bold,
};

export const h2 = {
  ...defaultTextStyles,
  fontSize: size.h2,
  fontFamily: fontFamily.bold,
};

export const h3 = {
  ...defaultTextStyles,
  fontSize: size.h3,
  fontFamily: fontFamily.light,
};

export const h4 = {
  ...defaultTextStyles,
  fontSize: size.h4,
};

export const body = {
  ...defaultTextStyles,
  fontSize: size.md,
};
