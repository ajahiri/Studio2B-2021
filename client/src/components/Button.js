import React from 'react';
import PropTypes from 'prop-types';
import {
  ActivityIndicator,
  StyleSheet,
  Text,
  TouchableHighlight,
} from 'react-native';

import { color, font, layout } from '../constants';

export default function Button({
  title,
  type,
  size,
  disabled,
  isLoading,
  onPress,
  ...props
}) {
  const sizeStyle = (size => {
    switch (size) {
      case 'small':
        return smallSizeStyle;
      case 'large':
      default:
        return largeSizeStyle;
    }
  })(size);

  const [colorStyle, focusedColor] = (type => {
    switch (type) {
      case 'primary':
        return [primaryColorStyle, color.accentFocused];
      case 'danger':
        return [dangerColorStyle, color.dangerFocused];
      default:
        return [secondaryColorStyle, color.lightGray];
    }
  })(type);

  return (
    <TouchableHighlight
      disabled={disabled}
      onPress={onPress}
      underlayColor={focusedColor}
      style={[
        sizeStyle.container,
        disabled ? colorStyle.disabled : colorStyle.default,
        props.style,
      ]}>
      {isLoading ? (
        <ActivityIndicator color={colorStyle.defaultText} />
      ) : (
        <Text
          style={[
            sizeStyle.title,
            disabled ? colorStyle.disabledText : colorStyle.defaultText,
          ]}>
          {title}
        </Text>
      )}
    </TouchableHighlight>
  );
}

Button.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['primary', 'secondary', 'danger']),
  size: PropTypes.oneOf(['large', 'small']),
  disabled: PropTypes.bool,
  isLoading: PropTypes.bool,
  onPress: PropTypes.func,
};

Button.defaultProps = {
  title: '',
  type: 'secondary',
  size: 'large',
  disabled: false,
  isLoading: false,
  onPress: () => {},
};

const commonContainerStyle = {
  alignItems: 'center',
  justifyContent: 'center',
  borderWidth: 2,
};

const largeSizeStyle = StyleSheet.create({
  container: {
    ...commonContainerStyle,
    height: 60,
    borderRadius: layout.radius.lg,
    borderWidth: 2,
    borderBottomWidth: 5,
  },
  title: {
    ...font.h3,
  },
});

const smallSizeStyle = StyleSheet.create({
  container: {
    ...commonContainerStyle,
    height: 40,
    borderRadius: layout.radius.md,
    borderBottomWidth: 5,
  },
  title: {
    ...font.largeBold,
  },
});

const defaultBorderColor = color.black;
const disabledBorderColor = color.gray;

const primaryColorStyle = StyleSheet.create({
  default: {
    backgroundColor: color.accent,
    borderColor: defaultBorderColor,
  },
  disabled: {
    backgroundColor: color.accentDisabled,
    borderColor: disabledBorderColor,
  },
  defaultText: {
    color: color.black,
  },
  disabledText: {
    color: color.gray,
  },
});

const secondaryColorStyle = StyleSheet.create({
  default: {
    backgroundColor: color.white,
    borderColor: defaultBorderColor,
  },
  disabled: {
    backgroundColor: color.white,
    borderColor: disabledBorderColor,
  },
  defaultText: {
    color: color.black,
  },
  disabledText: {
    color: color.gray,
  },
});

const dangerColorStyle = StyleSheet.create({
  default: {
    backgroundColor: color.danger,
    borderColor: defaultBorderColor,
  },
  disabled: {
    backgroundColor: color.dangerDisabled,
    borderColor: disabledBorderColor,
  },
  defaultText: {
    color: color.white,
  },
  disabledText: {
    color: color.lightGray,
  },
});
