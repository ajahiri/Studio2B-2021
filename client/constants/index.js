import * as color from './color';
import * as font from './font';
import * as layout from './layout';

export { color, font, layout };
