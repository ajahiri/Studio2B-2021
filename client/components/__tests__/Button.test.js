import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { toHaveStyle } from '@testing-library/jest-native';

import Button, {
  BUTTON_ACTIVITY_INDICATOR_ID,
  BUTTON_TEXT_ID,
  BUTTON_CONTAINER_ID,
} from '../Button';

import { color, font, layout } from '../../constants';

expect.extend({ toHaveStyle });

describe('Button', () => {
  const buttonTitle = 'My Button';

  describe('test large buttons', () => {
    describe('test large text buttons', () => {
      it('renders an enabled large text button', () => {
        const onPress = jest.fn();
        const { getByTestId, queryByTestId, toJSON } = render(
          <Button title={buttonTitle} onPress={onPress} />,
        );

        expect(queryByTestId(BUTTON_TEXT_ID)).not.toBeNull();
        expect(queryByTestId(BUTTON_ACTIVITY_INDICATOR_ID)).toBeNull();

        const buttonContainer = getByTestId(BUTTON_CONTAINER_ID);
        expect(buttonContainer).toHaveStyle({
          height: layout.buttonSizes.large,
          borderRadius: layout.radius.lg,
        });

        // Jest doesn't recognise that `RenderAPI.container` is disabled, so
        // instead we check the number of times our mock function has been
        // called. We do this for all test cases for Button.
        fireEvent.press(buttonContainer);
        expect(onPress).toHaveBeenCalledTimes(1);

        const buttonText = getByTestId(BUTTON_TEXT_ID);
        expect(buttonText.props.children).toBe(buttonTitle);

        expect(toJSON()).toMatchSnapshot();
      });

      it('renders a disabled large text button', () => {
        const onPress = jest.fn();
        const { getByTestId, queryByTestId, toJSON } = render(
          <Button disabled title={buttonTitle} onPress={onPress} />,
        );

        expect(queryByTestId(BUTTON_TEXT_ID)).not.toBeNull();
        expect(queryByTestId(BUTTON_ACTIVITY_INDICATOR_ID)).toBeNull();

        const buttonContainer = getByTestId(BUTTON_CONTAINER_ID);
        expect(buttonContainer).toHaveStyle({
          height: layout.buttonSizes.large,
          borderRadius: layout.radius.lg,
        });

        fireEvent.press(buttonContainer);
        expect(onPress).toHaveBeenCalledTimes(0);

        const buttonText = getByTestId(BUTTON_TEXT_ID);
        expect(buttonText.props.children).toBe(buttonTitle);

        expect(toJSON()).toMatchSnapshot();
      });
    });

    describe('test large loading buttons', () => {
      it('renders an enabled large loading button', () => {
        const onPress = jest.fn();
        const { getByTestId, queryByTestId, toJSON } = render(
          <Button isLoading title={buttonTitle} onPress={onPress} />,
        );

        expect(queryByTestId(BUTTON_TEXT_ID)).toBeNull();
        expect(queryByTestId(BUTTON_ACTIVITY_INDICATOR_ID)).not.toBeNull();

        const buttonContainer = getByTestId(BUTTON_CONTAINER_ID);
        expect(buttonContainer).toHaveStyle({
          height: layout.buttonSizes.large,
          borderRadius: layout.radius.lg,
        });

        fireEvent.press(buttonContainer);
        expect(onPress).toHaveBeenCalledTimes(0);

        expect(toJSON()).toMatchSnapshot();
      });

      it('renders a disabled large loading button', () => {
        const onPress = jest.fn();
        const { getByTestId, queryByTestId, toJSON } = render(
          <Button disabled isLoading title={buttonTitle} onPress={onPress} />,
        );

        expect(queryByTestId(BUTTON_TEXT_ID)).toBeNull();
        expect(queryByTestId(BUTTON_ACTIVITY_INDICATOR_ID)).not.toBeNull();

        const buttonContainer = getByTestId(BUTTON_CONTAINER_ID);
        expect(buttonContainer).toHaveStyle({
          height: layout.buttonSizes.large,
          borderRadius: layout.radius.lg,
        });

        fireEvent.press(buttonContainer);
        expect(onPress).toHaveBeenCalledTimes(0);

        expect(toJSON()).toMatchSnapshot();
      });
    });
  });

  describe('test small buttons', () => {
    describe('test small text buttons', () => {
      it('renders an enabled small text button', () => {
        const onPress = jest.fn();
        const { getByTestId, queryByTestId, toJSON } = render(
          <Button size="small" title={buttonTitle} onPress={onPress} />,
        );

        expect(queryByTestId(BUTTON_TEXT_ID)).not.toBeNull();
        expect(queryByTestId(BUTTON_ACTIVITY_INDICATOR_ID)).toBeNull();

        const buttonContainer = getByTestId(BUTTON_CONTAINER_ID);
        expect(buttonContainer).toHaveStyle({
          height: layout.buttonSizes.small,
          borderRadius: layout.radius.md,
        });

        // Jest doesn't recognise that `RenderAPI.container` is disabled, so
        // instead we check the number of times our mock function has been
        // called. We do this for all test cases for Button.
        fireEvent.press(buttonContainer);
        expect(onPress).toHaveBeenCalledTimes(1);

        const buttonText = getByTestId(BUTTON_TEXT_ID);
        expect(buttonText.props.children).toBe(buttonTitle);

        expect(toJSON()).toMatchSnapshot();
      });

      it('renders a disabled small text button', () => {
        const onPress = jest.fn();
        const { getByTestId, queryByTestId, toJSON } = render(
          <Button
            size="small"
            disabled
            title={buttonTitle}
            onPress={onPress}
          />,
        );

        expect(queryByTestId(BUTTON_TEXT_ID)).not.toBeNull();
        expect(queryByTestId(BUTTON_ACTIVITY_INDICATOR_ID)).toBeNull();

        const buttonContainer = getByTestId(BUTTON_CONTAINER_ID);
        expect(buttonContainer).toHaveStyle({
          height: layout.buttonSizes.small,
          borderRadius: layout.radius.md,
        });

        fireEvent.press(buttonContainer);
        expect(onPress).toHaveBeenCalledTimes(0);

        const buttonText = getByTestId(BUTTON_TEXT_ID);
        expect(buttonText.props.children).toBe(buttonTitle);

        expect(toJSON()).toMatchSnapshot();
      });
    });

    describe('test small loading buttons', () => {
      it('renders an enabled small loading button', () => {
        const onPress = jest.fn();
        const { getByTestId, queryByTestId, toJSON } = render(
          <Button
            isLoading
            size="small"
            title={buttonTitle}
            onPress={onPress}
          />,
        );

        expect(queryByTestId(BUTTON_TEXT_ID)).toBeNull();
        expect(queryByTestId(BUTTON_ACTIVITY_INDICATOR_ID)).not.toBeNull();

        const buttonContainer = getByTestId(BUTTON_CONTAINER_ID);
        expect(buttonContainer).toHaveStyle({
          height: layout.buttonSizes.small,
          borderRadius: layout.radius.md,
        });

        fireEvent.press(buttonContainer);
        expect(onPress).toHaveBeenCalledTimes(0);

        expect(toJSON()).toMatchSnapshot();
      });

      it('renders a disabled small loading button', () => {
        const onPress = jest.fn();
        const { getByTestId, queryByTestId, toJSON } = render(
          <Button
            disabled
            isLoading
            size="small"
            title={buttonTitle}
            onPress={onPress}
          />,
        );

        expect(queryByTestId(BUTTON_TEXT_ID)).toBeNull();
        expect(queryByTestId(BUTTON_ACTIVITY_INDICATOR_ID)).not.toBeNull();

        const buttonContainer = getByTestId(BUTTON_CONTAINER_ID);
        expect(buttonContainer).toHaveStyle({
          height: layout.buttonSizes.small,
          borderRadius: layout.radius.md,
        });

        fireEvent.press(buttonContainer);
        expect(onPress).toHaveBeenCalledTimes(0);

        expect(toJSON()).toMatchSnapshot();
      });
    });
  });
});

describe('Button styles', () => {
  const buttonTitle = 'My Button';

  describe('test primary button styles', () => {
    it('renders a primary default button ', () => {
      const { getByTestId } = render(
        <Button type="primary" title={buttonTitle} />,
      );

      const buttonContainer = getByTestId(BUTTON_CONTAINER_ID);
      expect(buttonContainer).toHaveStyle({
        backgroundColor: color.accent,
        borderColor: color.defaultBorderColor,
      });

      const buttonText = getByTestId(BUTTON_TEXT_ID);
      expect(buttonText).toHaveStyle({
        ...font.extraLargeBold,
        color: color.defaultDarkTextColor,
      });
    });

    it('renders a primary disabled button ', () => {
      const { getByTestId } = render(
        <Button type="primary" disabled title={buttonTitle} />,
      );

      const buttonContainer = getByTestId(BUTTON_CONTAINER_ID);
      expect(buttonContainer).toHaveStyle({
        backgroundColor: color.accentDisabled,
        borderColor: color.disabledBorderColor,
      });

      const buttonText = getByTestId(BUTTON_TEXT_ID);
      expect(buttonText).toHaveStyle({
        ...font.extraLargeBold,
        color: color.disabledDarkTextColor,
      });
    });
  });

  describe('test secondary button styles', () => {
    it('renders a secondary default button ', () => {
      const { getByTestId } = render(<Button title={buttonTitle} />);

      const buttonContainer = getByTestId(BUTTON_CONTAINER_ID);
      expect(buttonContainer).toHaveStyle({
        backgroundColor: color.white,
        borderColor: color.defaultBorderColor,
      });

      const buttonText = getByTestId(BUTTON_TEXT_ID);
      expect(buttonText).toHaveStyle({
        ...font.extraLargeBold,
        color: color.defaultDarkTextColor,
      });
    });

    it('renders a secondary disabled button ', () => {
      const { getByTestId } = render(<Button disabled title={buttonTitle} />);

      const buttonContainer = getByTestId(BUTTON_CONTAINER_ID);
      expect(buttonContainer).toHaveStyle({
        backgroundColor: color.white,
        borderColor: color.disabledBorderColor,
      });

      const buttonText = getByTestId(BUTTON_TEXT_ID);
      expect(buttonText).toHaveStyle({
        ...font.extraLargeBold,
        color: color.disabledDarkTextColor,
      });
    });
  });

  describe('test danger button styles', () => {
    it('renders a danger default button ', () => {
      const { getByTestId } = render(
        <Button type="danger" title={buttonTitle} />,
      );

      const buttonContainer = getByTestId(BUTTON_CONTAINER_ID);
      expect(buttonContainer).toHaveStyle({
        backgroundColor: color.danger,
        borderColor: color.defaultBorderColor,
      });

      const buttonText = getByTestId(BUTTON_TEXT_ID);
      expect(buttonText).toHaveStyle({
        ...font.extraLargeBold,
        color: color.defaultLightTextColor,
      });
    });

    it('renders a danger disabled button ', () => {
      const { getByTestId } = render(
        <Button disabled type="danger" title={buttonTitle} />,
      );

      const buttonContainer = getByTestId(BUTTON_CONTAINER_ID);
      expect(buttonContainer).toHaveStyle({
        backgroundColor: color.dangerDisabled,
        borderColor: color.disabledBorderColor,
      });

      const buttonText = getByTestId(BUTTON_TEXT_ID);
      expect(buttonText).toHaveStyle({
        ...font.extraLargeBold,
        color: color.disabledLightTextColor,
      });
    });
  });
});
