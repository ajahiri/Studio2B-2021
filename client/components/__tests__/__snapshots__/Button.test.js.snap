// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test large buttons test large loading buttons renders a disabled large loading button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAFAFA",
        "borderBottomWidth": 5,
        "borderColor": "#8F8F8F",
        "borderRadius": 16,
        "borderWidth": 2,
        "height": 60,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID="button-container"
>
  <ActivityIndicator
    animating={true}
    color="#141414"
    hidesWhenStopped={true}
    size="small"
    testID="button-activity-indicator"
  />
</View>
`;

exports[`Button test large buttons test large loading buttons renders an enabled large loading button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAFAFA",
        "borderBottomWidth": 5,
        "borderColor": "#141414",
        "borderRadius": 16,
        "borderWidth": 2,
        "height": 60,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID="button-container"
>
  <ActivityIndicator
    animating={true}
    color="#141414"
    hidesWhenStopped={true}
    size="small"
    testID="button-activity-indicator"
  />
</View>
`;

exports[`Button test large buttons test large text buttons renders a disabled large text button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAFAFA",
        "borderBottomWidth": 5,
        "borderColor": "#8F8F8F",
        "borderRadius": 16,
        "borderWidth": 2,
        "height": 60,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID="button-container"
>
  <Text
    style={
      Array [
        Object {
          "color": "#8F8F8F",
          "fontFamily": "Montserrat_700Bold",
          "fontSize": 24,
        },
      ]
    }
    testID="button-text"
  >
    My Button
  </Text>
</View>
`;

exports[`Button test large buttons test large text buttons renders an enabled large text button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAFAFA",
        "borderBottomWidth": 5,
        "borderColor": "#141414",
        "borderRadius": 16,
        "borderWidth": 2,
        "height": 60,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID="button-container"
>
  <Text
    style={
      Array [
        Object {
          "color": "#141414",
          "fontFamily": "Montserrat_700Bold",
          "fontSize": 24,
        },
      ]
    }
    testID="button-text"
  >
    My Button
  </Text>
</View>
`;

exports[`Button test small buttons test small loading buttons renders a disabled small loading button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAFAFA",
        "borderBottomWidth": 5,
        "borderColor": "#8F8F8F",
        "borderRadius": 12,
        "borderWidth": 2,
        "height": 40,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID="button-container"
>
  <ActivityIndicator
    animating={true}
    color="#141414"
    hidesWhenStopped={true}
    size="small"
    testID="button-activity-indicator"
  />
</View>
`;

exports[`Button test small buttons test small loading buttons renders an enabled small loading button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAFAFA",
        "borderBottomWidth": 5,
        "borderColor": "#141414",
        "borderRadius": 12,
        "borderWidth": 2,
        "height": 40,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID="button-container"
>
  <ActivityIndicator
    animating={true}
    color="#141414"
    hidesWhenStopped={true}
    size="small"
    testID="button-activity-indicator"
  />
</View>
`;

exports[`Button test small buttons test small text buttons renders a disabled small text button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAFAFA",
        "borderBottomWidth": 5,
        "borderColor": "#8F8F8F",
        "borderRadius": 12,
        "borderWidth": 2,
        "height": 40,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID="button-container"
>
  <Text
    style={
      Array [
        Object {
          "color": "#8F8F8F",
          "fontFamily": "Montserrat_700Bold",
          "fontSize": 18,
        },
      ]
    }
    testID="button-text"
  >
    My Button
  </Text>
</View>
`;

exports[`Button test small buttons test small text buttons renders an enabled small text button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FAFAFA",
        "borderBottomWidth": 5,
        "borderColor": "#141414",
        "borderRadius": 12,
        "borderWidth": 2,
        "height": 40,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID="button-container"
>
  <Text
    style={
      Array [
        Object {
          "color": "#141414",
          "fontFamily": "Montserrat_700Bold",
          "fontSize": 18,
        },
      ]
    }
    testID="button-text"
  >
    My Button
  </Text>
</View>
`;
